### PR TITLE
Initial pr

### PR DESCRIPTION
**Pull request by tinystarinagalaxy**

### CODE REVIEW

File: governance.go
Comment (DimaCantemir) [inline governance.go:1]: file is not needed...AFAIK it's related to Atlas

File: Dockerfile
Comment (DimaCantemir) [inline Dockerfile:1]: you have protobuf compilation docker image but no .proto file

File: utils/bx_client.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	return map[string]string{
+		"authorization": bc.authorization,
+	}, nil
+}
+
+func (bc blxrCredentials) RequireTransportSecurity() bool {
+	return false
+}
+
+// NewGRPCClient connects to custom Trader API
```
Comment (DimaCantemir) [inline utils/bx_client.go:51]: Trader API -> Intent Gateway

File: governance.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import (
+	"crypto/ecdsa"
+	"log"
+	"os"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+type Governance struct {
```
Comment (DimaCantemir) [inline governance.go:12]: Governance -> Solver. and this change needs to be reflected everywhere.

**Review by DimaCantemir — CHANGES_REQUESTED** [review]
Good work. A few changes recommended

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+IMAGE_NAME=bloxroute/bdn-protobuf:v3.19.3
+LOCAL_IMAGE_NAME=bx-proto-gen
+
+.PHONY: genproto
+.SILENT: genproto
+genproto:
+	docker run -v $(CURDIR):/go/protobuf --platform linux/amd64 $(IMAGE_NAME) \
+		protoc --go_out=. --go_opt=paths=source_relative  --go-grpc_out=. --go-grpc_opt=paths=source_relative gateway.proto
```
Comment (atulsriv) [inline Makefile:8]: are these paths correct? just verify that it works if the proto gens are copy/pasted

File: cmd/solver/main.go [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+func main() {
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt)
+	env, err := godotenv.Read(".env")
+	if err != nil {
+		log.Fatalf("could not load .env file: %s", err)
+	}
+	takeOrderRequestChan := make(chan entities.TakeOrderRequest, 10)
+	//go func() {
```
Comment (atulsriv) [inline cmd/solver/main.go:27]: delete this as well, don't think we need

File: cmd/solver/main.go [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	}
+
+	dappAddress := crypto.PubkeyToAddress(bundlerPrivateKey.PublicKey)
+
+	entities.NewAoriSolver(env["SOLVER_PK"], 5, dappAddress.String(), "AORI SOLVER", takeOrderRequestChan)
+	// CTRL + C to stop the program
+	<-interrupt
+}
+
+var authHeader = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ0eXBlIjoiQXBpS2V5IiwiZW1haWwiOiJkYW5pZWwudGFvQGJsb3hyb3V0ZS5jb20iLCJuYW1lIjoiRGFuaWVsIFRhbyIsInJhdGVMaW1pdCI6MTAwLCJpYXQiOjE3MDAwNzcyOTd9.O6STgok-EJ6IcCoQSpTZPp4sGJGe4ShZhz1rJLQ3fOI"
```
Comment (atulsriv) [inline cmd/solver/main.go:43]: I don't think we need this file anymore at all, can you delete it?

**Review by atulsriv — CHANGES_REQUESTED** [review]
just some minor comments

File: cmd/dapp/main.go [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+	// CTRL + C to stop the program
+	<-interrupt
+
+	close(shutdownChan)
+	close(makeOrderChan)
+
+}
+
+//
+//func main() {
```
[the remark this answers is not in the document]
Comment (tinystarinagalaxy) [reply cmd/dapp/main.go:33]: done

File: cmd/solver/main.go (hunk 4 above)
Comment (tinystarinagalaxy) [reply cmd/solver/main.go:27]: done

File: cmd/solver/main.go (hunk 5 above)
Comment (tinystarinagalaxy) [reply cmd/solver/main.go:43]: what do you mean, this is our solver, we need it for now, for the demo purposes and etc

File: governance.go
Comment (tinystarinagalaxy) [reply governance.go:1]: deleted this file

File: governance.go (hunk 2 above)
Comment (tinystarinagalaxy) [reply governance.go:12]: deleted this file

File: utils/bx_client.go (hunk 1 above)
Comment (tinystarinagalaxy) [reply utils/bx_client.go:51]: done

File: utils/bx_client.go [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+	Endpoint    string
+	DisableAuth bool
+	UseTLS      bool
+	AuthHeader  string
+}
+
+func DefaultRPCOpts(endpoint string) RPCOpts {
+	var ep string
+	if endpoint == "" {
+		ep = "localhost:5005"
```
[the remark this answers is not in the document]
Comment (tinystarinagalaxy) [reply utils/bx_client.go:28]: added

**Review by atulsriv — APPROVED** [review]

**Pull request merged**